src/Features/CSharp/Portable/CodeFixes/RemoveUnnecessaryCast/RemoveUnnecessaryCastCodeFixProvider.cs

```diff
@@ -21,89 +22,68 @@ namespace Microsoft.CodeAnalysis.CSharp.CodeFixes.RemoveUnnecessaryCast
-        {
-            var token = root.FindToken(span.Start);
-            if (!token.Span.IntersectsWith(span))
-            {
-                return null;
-            }
-
-            return token.GetAncestors<CastExpressionSyntax>()
-                .FirstOrDefault(c => c.Span.IntersectsWith(span) && c.IsUnnecessaryCast(model, cancellationToken));
-        }
+        public sealed override ImmutableArray<string> FixableDiagnosticIds { get; } =
+            ImmutableArray.Create(IDEDiagnosticIds.RemoveUnnecessaryCastDiagnosticId);
 
         public sealed override Task RegisterCodeFixesAsync(CodeFixContext context)
         {
-            context.RegisterCodeFix(
-                new MyCodeAction(
-                    CSharpFeaturesResources.Remove_Unnecessary_Cast,
-                    c => RemoveUnnecessaryCastAsync(context.Document, context.Span, c)),
+            context.RegisterCodeFix(new MyCodeAction(
+                FeaturesResources.Remove_Unnecessary_Cast,
+                c => FixAsync(context.Document, context.Diagnostics.First(), c)),
                 context.Diagnostics);
             return SpecializedTasks.EmptyTask;
         }
 
-        private static async Task<Document> RemoveUnnecessaryCastAsync(
-            Document document, TextSpan span, CancellationToken cancellationToken)
+        protected override async Task FixAllAsync(
+            Document document, ImmutableArray<Diagnostic> diagnostics,
+            SyntaxEditor editor, CancellationToken cancellationToken)
         {
-            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
-            var model = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
-            var cast = GetCastNode(root, model, span, cancellationToken);
-            return await RemoveUnnecessaryCastAsync(document, cast, cancellationToken).ConfigureAwait(false);
+            var castNodes = diagnostics.SelectAsArray(
+                d => (CastExpressionSyntax)d.AdditionalLocations[0].FindNode(getInnermostNodeForTie: true, cancellationToken));
+
+            await editor.ApplyExpressionLevelSemanticEditsAsync(
+                document, castNodes,
+                (semanticModel, castExpression) => castExpression.IsUnnecessaryCast(semanticModel, cancellationToken),
+                (_, currentRoot, castExpression) =>
+                {
+                    var (oldParent, newParent) = Update(castExpression);
+                    return currentRoot.ReplaceNode(oldParent, newParent);
+                },
+                cancellationToken).ConfigureAwait(false);
         }
 
-        private static async Task<Document> RemoveUnnecessaryCastAsync(
-            Document document, CastExpressionSyntax cast, CancellationToken cancellationToken)
+        private (ExpressionSyntax, ExpressionSyntax) Update(CastExpressionSyntax castExpression)
```

Comment:
:bulb: Prefer named tuple elements

Reply:
This method turned out to be fairly useless.  So i just inlined it :)